Sources/PackageGraph/ModuleAliasTracker.swift

```diff
@@ -131,219 +130,99 @@ class ModuleAliasTracker {
     // Propagate defined aliases upstream. If they are chained, the final
     // alias value will be applied
     func propagate(productID: String,
+                   observabilityScope: ObservabilityScope,
                    aliasBuffer: inout [String: ModuleAliasModel]) {
         let productAliases = aliasMap[productID] ?? []
         for aliasModel in productAliases {
             // Alias buffer is used to carry down aliases defined upstream
-            if aliasBuffer[aliasModel.name] == nil {
-                if let v = lookupAlias(key: aliasModel.alias, in: aliasBuffer),
-                   v != aliasModel.alias {
-                    let chainedModel = ModuleAliasModel(name: aliasModel.name, alias: v, originPackage: aliasModel.originPackage, consumingPackage: aliasModel.consumingPackage)
-                    aliasBuffer[aliasModel.name] = chainedModel
-                } else {
-                    aliasBuffer[aliasModel.name] = aliasModel
-                }
+            if let existing = aliasBuffer[aliasModel.name],
+               existing.alias != aliasModel.alias {
+                observabilityScope.emit(info: "Alias '\(aliasModel.alias)' for '\(aliasModel.name)' defined in '\(productID)' is overridden by '\(existing.alias)'; if this is not intended, remove the latter from 'moduleAliases' in its manifest")
```

Comment:
Do we have the ability to add some more info here? I'm thinking something like

> Alias 'X' for 'Y' defined in 'Product' of package 'Package' is overridden by 'Z' defined in 'OtherProduct' of package 'OtherPackage'; if this is not intended, remove the latter from 'moduleAliases' in its manifest

Reply:
It doesn't have access to 'OtherPackage/Product' right now but can add that in the future.